lib/util/global-env.ts

```diff
@@ -1,12 +1,14 @@
 // If we can't guarantee global state, don't fake it: fail instead.
-let globalEnv: undefined | typeof window | typeof self | typeof global;
+let globalEnv: typeof window | typeof self | typeof global;
 // Use window (web)/self (web worker)/global (node) as appropriate
 if (typeof window !== 'undefined') {
 	globalEnv = window;
 } else if (typeof self !== 'undefined') {
 	globalEnv = self;
 } else if (typeof global !== 'undefined') {
 	globalEnv = global;
+} else {
+	throw new Error('Could not get global env');
```

Comment:
This now matches the comment at the top and I think is safe as `setSharedOptions`/`fromSharedOptions` would both have blown up if we reached the previous `undefined` path